core/deployment/src/main/java/io/quarkus/deployment/index/ApplicationArchiveBuildStep.java

```diff
@@ -171,13 +172,12 @@ static Path urlToPath(URL url) {
                 String jarPath = url.getPath().substring(0, url.getPath().lastIndexOf('!'));
                 return Paths.get(new URI(jarPath));
             } else if (url.getProtocol().equals("file")) {
-                int index = url.getPath().lastIndexOf("/META-INF");
+                final String path = Paths.get(url.toURI()).toString();
+                final int index = path.lastIndexOf(File.separator + "META-INF");
                 if (index == -1) {
-                    return Paths.get(url.getPath());
+                    return Paths.get(path);
```

Comment:
We could avoid recreating the path here, maybe?

Reply:
Done